### ISSUE TITLE
Possible wrong version of tensorflow in DeepCoy-env.yml

### ISSUE THREAD

**Issue by miles-mcgibbon** [fimrie/DeepCoy#2]
Hi,

When running the examples to generate molecules with the pretrained models, or running the example Jupyter Notebook (both in a newly made conda environment setup with the DeepCoy-env.yml file) I get a tensorflow error:

Input:
```
(DeepCoy-env) XXXX@XXXX:~/DeepCoy$ python3 DeepCoy.py --restore models/DeepCoy_DUDE_model_e09.pickle --dataset zinc --config '{"generation": true, "number_of_generation_per_valid": 1000, "batch_size": 1, "train_file": "data/molecules_zinc_decoys_valid.json", "valid_file": "data/molecules_zinc_dekois_valid.json", "output_name": "output/DeepCoy_generated_decoys_zinc_dekois_valid.txt"}'
```

Output:
```
Traceback (most recent call last):
  File "DeepCoy.py", line 1129, in <module>
    model = DenseGGNNChemModel(args)
  File "DeepCoy.py", line 55, in __init__
    super().__init__(args)
  File "/DeepCoy/GGNN_DeepCoy.py", line 80, in __init__
    self.make_model()
  File "/DeepCoy/GGNN_DeepCoy.py", line 126, in make_model
    self.prepare_specific_graph_model()
  File "DeepCoy.py", line 196, in prepare_specific_graph_model
    cell = tf.nn.rnn_cell.DropoutWrapper(cell,
AttributeError: module 'tensorflow.python.ops.nn' has no attribute 'rnn_cell'
```

When trying to debug, it looks as though tf.nn.rnn_cell was moved into tf.contrib from version 1.0 (this contrib usage seems to be used in line 195 and 209 of DeepCoy also?)

Changing lines 195, 196 and 209, 210 from:
```
cell = tf.contrib.rnn.GRUCell(new_h_dim)
cell = tf.nn.rnn_cell.DropoutWrapper(cell,
```

to 

```
cell = tf.contrib.rnn.GRUCell(new_h_dim)
cell = tf.contrib.rnn.DropoutWrapper(cell,
```

Fixes this particular error, but then the below error is produced from the same input:
Output:
```
File "DeepCoy.py", line 1129, in <module>
    model = DenseGGNNChemModel(args)
  File "DeepCoy.py", line 55, in __init__
    super().__init__(args)
  File "/DeepCoy/GGNN_DeepCoy.py", line 80, in __init__
    self.make_model()
  File "/DeepCoy/GGNN_DeepCoy.py", line 142, in make_model
    "_encoder")
  File "DeepCoy.py", line 307, in compute_final_node_representations_with_residual
    h = self.weights['node_gru'+scope_name+str(iter_idx)](acts, h)[1]                       # [b*v, h]
  File "/.local/lib/python3.6/site-packages/tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py", line 713, in __call__
    output, new_state = self._cell(inputs, state, scope)
  File /.local/lib/python3.6/site-packages/tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py", line 150, in __call__
    [inputs, state], 2 * self._num_units, True, 1.0))
  File "/.local/lib/python3.6/site-packages/tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py", line 1044, in _linear
    _WEIGHTS_VARIABLE_NAME, [total_arg_size, output_size], dtype=dtype)
  File "/.local/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 1049, in get_variable
    use_resource=use_resource, custom_getter=custom_getter)
  File "/.local/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 948, in get_variable
    use_resource=use_resource, custom_getter=custom_getter)
  File "/.local/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 356, in get_variable
    validate_shape=validate_shape, use_resource=use_resource)
  File "/.local/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 341, in _true_getter
    use_resource=use_resource)
  File "/.local/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py", line 653, in _get_single_variable
    name, "".join(traceback.format_list(tb))))
ValueError: Variable graph_model/gru_scope_encoder0/gru_cell/gates/weights already exists, disallowed. Did you mean to set reuse=True in VarScope? Originally defined at:

  File "/.local/lib/python3.6/site-packages/tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py", line 1044, in _linear
    _WEIGHTS_VARIABLE_NAME, [total_arg_size, output_size], dtype=dtype)
  File "/.local/lib/python3.6/site-packages/tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py", line 150, in __call__
    [inputs, state], 2 * self._num_units, True, 1.0))
  File "/.local/lib/python3.6/site-packages/tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py", line 713, in __call__
    output, new_state = self._cell(inputs, state, scope)

> /.local/lib/python3.6/site-packages/tensorflow/python/ops/variable_scope.py(653)_get_single_variable()
-> name, "".join(traceback.format_list(tb))))
```

**Comment by fimrie** [comment]
Hi,

Thanks for your interest in our work. We are aware that some functionality of tf.nn.rnn_cell was moved to tf.contrib but I believe this occurred after 1.10. On my system the code runs with Tensorflow 1.10, but requires fix you mention when moving to Tensorflow 1.14 (for example). 

Please could you check the version of Tensorflow you're using? The below command should do this.
```python3 -c 'import tensorflow as tf; print(tf.__version__)'```

**Comment by miles-mcgibbon** [comment]
Hi,

Thank you for your quick reply. Here is my output:

```
python3 -c 'import tensorflow as tf; print(tf.__version__)'
/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:455: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint8 = np.dtype([("qint8", np.int8, 1)])
/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:456: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint8 = np.dtype([("quint8", np.uint8, 1)])
/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:457: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint16 = np.dtype([("qint16", np.int16, 1)])
/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:458: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_quint16 = np.dtype([("quint16", np.uint16, 1)])
/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:459: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  _np_qint32 = np.dtype([("qint32", np.int32, 1)])
/.local/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:462: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.
  np_resource = np.dtype([("resource", np.ubyte, 1)])
1.1.0
```

After getting this result, I tried running DeepCoy.py again in a new VM on a remote server using the supplied conda environment and everything worked perfectly. I think it must have been a problem with my local python path variables in my .bashrc, and nothing to do with your code - thank you again for your time and apologies for the issue on my end. I thoroughly enjoyed reading your publication and think this decoy generation method is a phenomenal solution to biased existing decoy sets. 

I will close this issue with this comment - many thanks again.

Miles
